classes/services/queryBuilders/UserListQueryBuilder.inc.php

```diff
@@ -276,15 +309,30 @@ public function get() {
 		$q = Capsule::table('users as u')
 					->leftJoin('user_user_groups as uug', 'uug.user_id', '=', 'u.user_id')
 					->leftJoin('user_groups as ug', 'ug.user_group_id', '=', 'uug.user_group_id')
-					->where('ug.context_id','=', $this->contextId)
 					->orderBy($this->orderColumn, $this->orderDirection)
 					->groupBy('u.user_id');
 
+		// context
+		if (!empty($this->contextIds)) {
+			$q->whereIn('ug.context_id', $this->contextIds);
+		} elseif (!empty($this->contextId)) {
+			$q->where('ug.context_id', '=', $this->contextId);
+		}
+
 		// roles
 		if (!is_null($this->roleIds)) {
 			$q->whereIn('ug.role_id', $this->roleIds);
 		}
 
+		// user groups
+		if (!is_null($this->userGroupIds)) {
+			if (in_array(0, $this->userGroupIds)) {
+				$q->whereNull('ug.user_group_id');
```

Comment:
hmmmm.... what does this mean/make?

Reply:
Pass 0 to return users with no groups.

Reply:
hmm... do you maybe have the SQL? -- it looks to me that it would have then something like "WHERE ug.user_group_id IS NULL", or? -- wouldn't then in that case all user assignments be returned?
If I see it correctly, earlier, the contextId was set to null in this function https://github.com/pkp/pkp-lib/blob/master/classes/security/UserGroupDAO.inc.php#L491, to get the users witch no role. Is that then the same... Hmmm...
I will install this locally -- wanted to test it however -- then I can see it better, I think :-)

Reply:
Yes, I think the SQL will be `WHERE ug.user_group_id IS NULL`. My understanding is that this will return any users which do not have _any_ matching row in `ug`.

Reply:
Yes, I think you are right! :+1: